src/components/HeadCellMenu/MenuList/MenuGroup.tsx

```diff
@@ -71,11 +71,9 @@ const MenuGroupItems = ({
         onClick={handleOnClick}
         disabled={!enabled}
         autoFocus={autoFocus}
-        onKeyDown={handleHeadCellMenuKeyDown}
-        // TODO:
-        // implement onKeyDown with handleHeadCellMenuKeyDown for sn-table
         isSubMenu={!!isSubMenu}
         isActive={!!isActive}
+        {...(handleHeadCellMenuKeyDown ? { onKeyDown: handleHeadCellMenuKeyDown } : {})}
```

Comment:
Did you get type issue if you just passed `onKeyDown={handleHeadCellMenuKeyDown}`?